### PR TITLE
merge: Parser -> main added parsing functionality

### PR DESCRIPTION
**Pull request by josterhu**

### CODE REVIEW

File: back-end/java/main/java/src/Parsing/EnumUtils.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package src.Parsing;
+
+import src.DormRoom.BathroomType;
+import src.DormRoom.RoomCapacity;
+
+import java.io.IOException;
+
+public class EnumUtils {
```
Comment (dbchanin) [inline back-end/java/main/java/src/Parsing/EnumUtils.java:8]: Please rename this class to something more readable, such as ParserUtils

File: back-end/java/main/java/src/Parsing/EnumUtils.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
+package src.Parsing;
```
Comment (dbchanin) [inline back-end/java/main/java/src/Parsing/EnumUtils.java:1]: This class needs to be javadocked.

File: back-end/java/main/java/src/Parsing/EnumUtils.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        }
+
+        throw new IllegalArgumentException("Unexpected room type" + size + size.length());
+    }
+
+    public static boolean parseYesNo(String input) throws IllegalArgumentException {
+        return switch (input) {
+            case "Yes" -> true;
+            case "No" -> false;
+            default -> throw new IllegalArgumentException("Input must be Yes or No");
```
Comment (dbchanin) [inline back-end/java/main/java/src/Parsing/EnumUtils.java:50]: Should be a null, default for the last switch case.

File: back-end/java/main/java/src/Parsing/Parser.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
+package src.Parsing;
```
Comment (dbchanin) [inline back-end/java/main/java/src/Parsing/Parser.java:1]: This class needs to be javadocked.

File: back-end/java/main/java/src/Parsing/RoomParser.java [hunk 5]
```diff
@@ -0,0 +1,153 @@
+package src.Parsing;
```
Comment (dbchanin) [inline back-end/java/main/java/src/Parsing/RoomParser.java:1]: Add javadocks

File: back-end/java/main/java/src/Parsing/Parser.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+import java.io.IOException;
+import java.io.Reader;
+import java.util.Iterator;
+
+
+
+
+public class Parser {
+
+    public CSVParser csvParser;
```
Comment (dbchanin) [inline back-end/java/main/java/src/Parsing/Parser.java:24]: Do those fields need to be public? Could they also be final?

File: back-end/java/main/java/src/Parsing/RoomParser.java [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+
+import org.apache.commons.csv.CSVRecord;
+import src.DormRoom.*;
+
+public class RoomParser {
+
+    public Parser parser;
```
Comment (dbchanin) [inline back-end/java/main/java/src/Parsing/RoomParser.java:13]: Should parser be private final?

File: back-end/java/main/java/src/Parsing/RoomParser.java [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+
+    public RoomParser(String path) throws FileNotFoundException, IOException {
+        this.parser = new Parser(path);
+    }
+
+    public ArrayList<IDormRoom> getRooms(){
+        ArrayList<IDormRoom> result = new ArrayList<>();
+
+
```
Comment (dbchanin) [inline back-end/java/main/java/src/Parsing/RoomParser.java:22]: 2 lines of space between this is not needed. Could just be one line. I know this is pedantic.

File: back-end/java/main/java/src/Parsing/RoomParser.java [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+public class RoomParser {
+
+    public Parser parser;
+
+    public RoomParser(String path) throws FileNotFoundException, IOException {
+        this.parser = new Parser(path);
+    }
+
+    public ArrayList<IDormRoom> getRooms(){
```
Comment (dbchanin) [inline back-end/java/main/java/src/Parsing/RoomParser.java:19]: Exceptions listed in the below comments should be added to the method signature.

File: back-end/java/main/java/src/Parsing/RoomParser.java [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+            CSVRecord room = roomsIterator.next();
+
+            // Consecutive suite case
+            if (temp != null) {
+                // List to keep track of dorm rooms that belong to this suite
+                ArrayList<DormRoom> roomsInSuite = new ArrayList<>();
+                roomsInSuite.add(parseDormRoom(temp));
+                String suiteNumber = temp.get("Suite");
+
+                int commonAreaSize = Integer.parseInt(temp.get("Common Area Size"));
```
Comment (dbchanin) [inline back-end/java/main/java/src/Parsing/RoomParser.java:38]: This throws an exception for an invalid input, such as ":)" for parseInt?

File: back-end/java/main/java/src/Parsing/RoomParser.java (hunk 9 above)
Comment (dbchanin) [inline back-end/java/main/java/src/Parsing/RoomParser.java:19]: This method is pretty long. Please consider dividing this into more helper functions. Additionally, I see the potential for NullPointerExceptions, if the dataset was not properly constructed. While the dataset you made is beautiful and perfect ❤️ , I think a NullPointer should be handled if the dataset was accidentally tampered with.

**Review by dbchanin — COMMENTED** [review]
Overall, this code is great! It is pretty readable and high quality!

I made some minor pedantic comments across different methods. Additionally, Javadocs needs to be written for all classes and methods.

File: back-end/java/main/java/src/Parsing/RoomParser.java (hunk 7 above)
Comment (josterhu) [reply back-end/java/main/java/src/Parsing/RoomParser.java:13]: should be public final, ty

**Pull request closed without merging**